packages/babel-generator/src/buffer.js

```diff
@@ -116,30 +116,50 @@ export default class Buffer {
     filename: ?string,
     force?: boolean,
   ): void {
-    // If there the line is ending, adding a new mapping marker is redundant
-    if (this._map && str[0] !== "\n") {
-      this._map.mark(
-        this._position.line,
-        this._position.column,
-        line,
-        column,
-        identifierName,
-        filename,
-        force,
-      );
-    }
-
     this._buf.push(str);
     this._last = str[str.length - 1];
 
-    for (let i = 0; i < str.length; i++) {
-      if (str[i] === "\n") {
-        this._position.line++;
-        this._position.column = 0;
-      } else {
-        this._position.column++;
+    let last = 0;
+    let i = str.indexOf("\n");
```

Comment:
Should we use `/\r\n?|[\n\u2028\u2029]/` for spec compliance? https://tc39.es/ecma262/#prod-LineTerminatorSequence

Reply:
Possibly for `\u2028` and `\u2029`, but I opted not to because they're soo uncommon and doing a regex search would be much slower.

But `\r` and `\r\n` don't actually exist in strings, only `\n`. When we parse, we normalize them all to `\n`. In runtime, it's unobservable: ``` eval('`\r`') ``` (using eval to cause a parse of the string) is equivalent to `"\n"`.

Reply:
Can you leave a comment here that we intentionally ignore `\u2028` and `\u2029` for performance concerns?

Reply:
Done.